### DISCUSSION TITLE
Modifying the Runtime and RenderMode in appsettings.json file in iis folder

### DISCUSSION THREAD

**Discussion by engsky** [oqtane/oqtane.framework#2586] [Q&A] [answered]
I Modify the Runtime and RenderMode in appsettings.json file and making Restart Application after restarting no change in browser?! Why there is no change although the Runtime and RenderMode taken from appsettings.json in startup  ?
How to restart the application with the new configuration? Please give us discus @sbwalker

**Comment by sbwalker** [comment, marked as answer]
Runtime and Render Mode are Site Settings which can be specified independently for each site in your installation - the appsettings.json settings are only used as default values during the initial installation and when creating new sites. If you want to change Runtime or Render Mode you need to login and navigate to Admin Dashboard then Site Settings and scroll down to the Hosting Model section in the UI.

**Reply by engsky** [reply]
@sbwalker i didnt see assign the runtime from site settings. Ok if i need to restart the application after some configuration changes how can i make restarting the entire application re registering the ID services? I need that for some configuration about Online offline mode.

**Reply by sbwalker** [reply]
Oqtane uses IHostApplicationLifetime.StopApplication() to restart the framework - which can be invoked via the API GET method "api/Installation/restart" - however the method is only available to Host users (for obvious reasons).

**Reply by engsky** [reply]
In general, Is It possible restarting the entire application reregistering the ID services after configuration modification?

**Reply by sbwalker** [reply]
I am not sure what you are asking. I assume when you say "ID services" you are referring to Dependency Injection (DI)... and the answer is that yes, you must restart the application if you make any changes which require registering new services in DI (this is because .NET can only register services at startup). Your options for restarting the application will depend on your environment. There is the interactive method I described above where you need to be logged in as a Host user. If you are unable to login then you could create your own custom module with an anonymous API endpoint which calls IHostApplicationLifetime.StopApplication() - but you would do so at your own risk, as if anyone ever discovered this endpoint they could call it remotely to force a site restart.
